std/typecons.d

```diff
@@ -5189,6 +5189,8 @@ mixin template Proxy(alias a)
                 return a.opCmp(b);
             else static if (is(typeof(b.opCmp(a))))
                 return -b.opCmp(a);
+            else static if (isFloatingPoint!ValueType || isFloatingPoint!B)
+                return a < b ? -1 : a > b ? +1 : a == b ? 0 : float.nan;
             else
                 return a < b ? -1 : a > b ? +1 : 0;
```

Comment:
This may seem minor, but while at it could you please rewrite this as `return a < b ? -1 : (a > b)`. The fact that the second branch computes a Boolean without any additional processing is key to generating good code for this. Consider http://goo.gl/VY2XLo, which show that the seemingly equivalent expressions `return a > b ? 1 : a < b ? -1 : 0;` and `return a == b ? 0 : a > b ? +1 : -1;` both generate branches.

If you have the time, I also suggest you benchmark `return (a > b) - (a < b);`, which generates one extra instruction but seemingly smaller code.